jerry-core/parser/js/js-scanner.c

```diff
@@ -829,10 +829,8 @@ scanner_scan_primary_expression_end (parser_context_t *context_p, /**< context *
 
       lexer_next_token (context_p);
 
-      if (binding_type == SCANNER_BINDING_CATCH)
+      if (binding_type == SCANNER_BINDING_CATCH && context_p->stack_top_uint8 == SCAN_STACK_CATCH_STATEMENT)
       {
-        JERRY_ASSERT (context_p->stack_top_uint8 == SCAN_STACK_CATCH_STATEMENT);
```

Comment:
I don't understand this. Shouldn't we have different stack top for nested bindings?

Reply:
Let's see the original example:

`catch([{}])`. The type of the both `[]` and `{}` patterns are `SCANNER_BINDING_CATCH`. 

The stack top for the outer `[]` pattern is `SCAN_STACK_CATCH_STATEMENT`, but the stack top for the inner `{}` is `SCAN_STACK_ARRAY_LITERAL`.

Reply:
Hm, what about this:
`catch([{a} = { a:5 }])`

Reply:
There is no difference, since the scanner context's binding type is set in the beginning of the catch header, therefore the inner `{a} = { a:5 }` binding parsed instantly as destructuring pattern. So when the inner `{}` is finalized, the stack top is still `SCAN_STACK_ARRAY_LITERAL`.